jujugui/static/gui/src/app/init/component-renderers-mixin.js

```diff
@@ -715,58 +717,43 @@ Browser: ${navigator.userAgent}`
     ReactDOM.render(
       <Account
         acl={this.acl}
-        addAddress={
-          this.payment && this.payment.addAddress.bind(this.payment)}
-        addBillingAddress={
-          this.payment && this.payment.addBillingAddress.bind(this.payment)}
         addNotification={this._bound.addNotification}
+        controllerAPI={shapeup.addReshape({
+          getCloudCredentialNames: controllerAPI.getCloudCredentialNames.bind(controllerAPI),
+          listClouds: controllerAPI.listClouds.bind(controllerAPI),
+          revokeCloudCredential: controllerAPI.revokeCloudCredential.bind(controllerAPI),
+          updateCloudCredential: controllerAPI.updateCloudCredential.bind(controllerAPI)
+        })}
         controllerIsReady={this._controllerIsReady.bind(this)}
-        createCardElement={
-          this.stripe && this.stripe.createCardElement.bind(this.stripe)}
-        createPaymentMethod={
-          this.payment && this.payment.createPaymentMethod.bind(this.payment)}
-        createToken={this.stripe && this.stripe.createToken.bind(this.stripe)}
-        createUser={
-          this.payment && this.payment.createUser.bind(this.payment)}
-        generateCloudCredentialName={
-          initUtils.generateCloudCredentialName}
-        getUser={this.payment && this.payment.getUser.bind(this.payment)}
-        getCharges={
-          this.payment && this.payment.getCharges.bind(this.payment)}
-        getCloudCredentialNames={
-          controllerAPI.getCloudCredentialNames.bind(controllerAPI)}
-        getCloudProviderDetails={
-          initUtils.getCloudProviderDetails.bind(
-            initUtils)}
-        getCountries={
-          this.payment && this.payment.getCountries.bind(this.payment)}
-        getReceipt={
-          this.payment && this.payment.getReceipt.bind(this.payment)}
-        listClouds={controllerAPI.listClouds.bind(controllerAPI)}
-        removeAddress={
-          this.payment && this.payment.removeAddress.bind(this.payment)}
-        removeBillingAddress={
-          this.payment && this.payment.removeBillingAddress.bind(
-            this.payment)}
-        removePaymentMethod={
-          this.payment && this.payment.removePaymentMethod.bind(this.payment)}
-        revokeCloudCredential={
-          controllerAPI.revokeCloudCredential.bind(controllerAPI)}
+        initUtils={shapeup.addReshape({
+          generateCloudCredentialName: initUtils.generateCloudCredentialName.bind(initUtils),
+          getCloudProviderDetails: initUtils.getCloudProviderDetails.bind(initUtils),
+          validateForm: initUtils.validateForm.bind(initUtils)
+        })}
+        payment={payment && shapeup.addReshape({
```

Comment:
Can't you just use shapeup.fromShape here?

Reply:
Also above in the other similar changes?

Reply:
Ah cool, very handy.